types/firefox-webext-browser/index.d.ts

```diff
@@ -0,0 +1,3128 @@
+// Type definitions for WebExtension Development in FireFox 58.0
+// Project: https://developer.mozilla.org/en-US/Add-ons/WebExtensions
+// Definitions by: Jacob Bom <https://github.com/bomjacob>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.4
+// Generated using script at github.com/bomjacob/definitelytyped-firefox-webext-browser
+
+interface EventListener<T extends (...args: any[]) => any> {
+    addListener: (callback: T) => void;
+    removeListener: (listener: T) => void;
+    hasListener: (listener: T) => boolean;
+}
+
+declare namespace browser.alarms {
+    /* alarms types */
+    interface Alarm {
+        name: string;
+        scheduledTime: number;
+        periodInMinutes?: number;
+    }
+
+    /* alarms functions */
+    function create(alarmInfo: {
+        when?: number;
+        delayInMinutes?: number;
+        periodInMinutes?: number;
+    }): void;
+    function create(name: string, alarmInfo: {
+        when?: number;
+        delayInMinutes?: number;
+        periodInMinutes?: number;
+    }): void;
+
+    function get(name?: string): Promise<Alarm>;
+
+    function getAll(): Promise<Alarm[]>;
+
+    function clear(name?: string): Promise<boolean>;
+
+    function clearAll(): Promise<boolean>;
+
+    /* alarms events */
+    const onAlarm: EventListener<(name: Alarm) => void>;
+}
+
+declare namespace browser.manifest {
```

Comment:
I assume you mean that there is no `browser.manifest` namespace exposed?

I've renamed it to _manifest for now...
I've tried to see if there is any way to make some functions or types "private" so that they wouldn't show up in autocomplete, but I've found nothing so far.